### ISSUE TITLE
local application binary is deleted after cloudfoundry_app resource update

### ISSUE THREAD

**Issue by abendt** [cloudfoundry-community/terraform-provider-cf#198]
my local application binary is removed after i run _terraform apply_ successfully. the application is correctly pushed to CF. 

```
resource "cloudfoundry_app" "sba-cl" {
  name = "sba-cl"
  space = "${data.cloudfoundry_space.space.id}"
  url = "file:///Users/abendt/myProject/terraform/build/terraform/app/app.jar"

  environment {
    JAVA_OPTS = "-Djava.security.egd=file:/dev/./urandom"
    SPRING_PROFILES_ACTIVE = "sandbox"
    JBP_CONFIG_OPEN_JDK_JRE = "{jre: { version: 11.+ }}"
  }
}
```

i'd expect the plugin to not remove my local file

**Comment by gberche-orange** [comment]
We need to further investigate / reproduced as briefly at the code at https://github.com/mevansam/terraform-provider-cf/blob/88f08f637ddc475dfec7a4dcb583b2d1f9c5e61d/cloudfoundry/resource_cf_app.go#L475-L500 care was taken to exclude file:// urls.

**Comment by abendt** [comment]
@gberche-orange i tried to recompile the plugin locally to add some debug output.
first i got confused as i could not reproduce the issue, then i noticed  that the addRessource case actually works as expected. 
However the updateRessource case does not seem to have a check for a local file:// URL (see 
https://github.com/mevansam/terraform-provider-cf/blob/88f08f637ddc475dfec7a4dcb583b2d1f9c5e61d/cloudfoundry/resource_cf_app.go#L968)

could this be the reason?

**Comment by abendt** [comment]
this change fixes the issue for me: https://github.com/abendt/terraform-provider-cf/commit/95061fd0cdf902e16b5cd7713163e432aa23f7b1

**Comment by gberche-orange** [comment]
thanks @abendt !

Would you be willing to send a PR with this fix, potentially with acceptance test refinement to assert the local file artefact isn't deleted in both initial provisionning and update ?

https://github.com/mevansam/terraform-provider-cf/blob/88f08f637d/cloudfoundry/resource_cf_app_test.go#L50-L60

**Comment by abendt** [comment]
@gberche-orange sure i can do that. i can also have a look at the acceptance tests. i might however have some further questions there about setup, etc. as i am a golang newbie ...

**Comment by abendt** [comment]
@gberche-orange 

i'm afraid i need some more guidance here:

* https://github.com/mevansam/terraform-provider-cf#developing-the-provider

when i run _make build_ i get an error that gometalinter.v2 is missing. could you provide more detailed installations instructions here how to properly setup the project for development?
I was able to get around that by directly invoking _go install_ so this does not seem to be a blocker.

* https://github.com/mevansam/terraform-provider-cf#testing-the-provider

i tried to use https://docs.pivotal.io/pcf-dev/index.html
set the environment variables:

```
export CF_API_URL=https://api.dev.cfdev.sh
export CF_USER=user
export CF_PASSWORD=pass
export CF_CA_CERT=""
export CF_SKIP_SSL_VALIDATION=true
export CF_UAA_CLIENT_ID=admin
export CF_UAA_CLIENT_SECRET=admin
```

where do i get the correct values for UAA_CLIENT_ID and UAA_CLIENT_SECRET from?

when i run _TF_ACC=1 go test -v -timeout 120m ._ i get an error stating that the environment variables GITHUB_USER and GITHUB_TOKEN need to be set too. what are they used for?

**Comment by gberche-orange** [comment]
> I was able to get around that by directly invoking go install so this does not seem to be a blocker.

Great, also travis.yml and https://github.com/mevansam/terraform-provider-cf/blob/dev/scripts/prepare-tfacc-env.sh might be a good source of inspiration

https://github.com/mevansam/terraform-provider-cf/blob/f03f28fe8a8c73db847275ff716da1f1a0e71da1/.travis.yml#L7-L16

> where do i get the correct values for UAA_CLIENT_ID and UAA_CLIENT_SECRET from?

This is for testing the user creation https://github.com/mevansam/terraform-provider-cf/blob/dev/website/docs/index.html.markdown https://github.com/mevansam/terraform-provider-cf/blob/dev/website/docs/r/user.html.markdown

I guess you need to create a UAA client id and client_secret in PCF-dev. I'm not sure how to retreive the uaa admin user to do so. @psycofdj @ArthurHlt @janosbinder  Any hint ?

https://github.com/pivotal-cf/pcfdev/blob/fb24b9ce9320de6820c1c890bfa1bc4c518e74f7/src/api/usecases/uaa_credential_replacement_test.go


> i get an error stating that the environment variables GITHUB_USER and GITHUB_TOKEN need to be set too. what are they used for?

This is for testing the cloudfoundry_app.git support https://github.com/mevansam/terraform-provider-cf/blob/dev/website/docs/r/app.html.markdown#application-source--binary

**Comment by ArthurHlt** [comment]
On rework inside #203 this can't occurred has this behaviour has been removed.

@gberche-orange i let you decide if we close this issue or do we handle it in current provider

**Comment by gberche-orange** [comment]
thanks @ArthurHlt. Lets keep this issue open until #203 gets merged to inform users about the current limitation. I suggest to not spend effort trying ti fix it outside of #203 for now.

**Comment by ArthurHlt** [comment]
Please check new release at https://github.com/cloudfoundry-community/terraform-provider-cf/releases/tag/v0.10.0 to see if it fix your issue.
I'm closing it for now, please re-open if you still found on issue on this.
